desc/objectives/utils.py

```diff
@@ -185,11 +170,12 @@ def factorize_linear_constraints(objective, constraint, x_scale="auto"):  # noqa
     # null space & particular solution
     A = A * D[None, unfixed_idx]
     if A.size:
-        A_inv, Z = svd_inv_null(A)
+        x_p, Z = qr_inv_null(A, b)
+        xp = put(xp, unfixed_idx, x_p)
     else:
         A_inv = A.T
         Z = np.eye(A.shape[1])
-    xp = put(xp, unfixed_idx, A_inv @ b)
+        xp = put(xp, unfixed_idx, A_inv @ b)
```

Comment:
or if there is a reason to move it, what is the reason?

Reply:
With SVD it is easier to calculate `A_inv` but with QR, it is easier to solve Rx=b kind of system. So, instead of returning `A_inv` the new function `qr_xp_null` returns the particular solution and the null-space. I could also return `Q, R` and calculate `x_p` here, but it is more convenient to calculate it in `qr_xp_null`.